File: src/rb/file_system.rb
folders = Array.new
files = Array.new

folderPath = File.join(Dir.getwd, "../files")
Dir.foreach(folderPath) do |childItem|
    if childItem == "." || childItem == ".."
        next
    end

    if File.directory?(File.join(folderPath, childItem))
        folders.push(childItem)
    end
    if File.file?(File.join(folderPath, childItem))
        files.push(childItem)
    end
end

folders.each{ |folder| puts "/#{folder}" }

files.each() do |file|
    fileSize = File.new(File.join(folderPath, file)).stat.size
    puts "#{file},#{fileSize}B"
end

File: src/php/file_system.php
<?php
$folderPath = __DIR__ . "/../files/";

$folders = array();
$files = array();

foreach(scandir($folderPath) as $childItem)
{
    if ($childItem == "." || $childItem == "..")
        continue;

    if (is_dir($folderPath . $childItem))
        array_push($folders, $childItem);
    if (is_file($folderPath . $childItem))
        array_push($files, $childItem);
}

foreach($folders as $folder){
    echo "/$folder\n";
}

foreach($files as $file){
    $fileSize = filesize($folderPath . $file);
    echo "$file,$fileSize". "B\n";
}
?>